libs/libarchfpga/src/arch_util.cpp

```diff
@@ -643,30 +637,29 @@ void ProcessLutClass(t_pb_type* lut_pb_type) {
     lut_pb_type->modes[1].parent_pb_type = lut_pb_type;
     lut_pb_type->modes[1].index = 1;
     lut_pb_type->modes[1].num_pb_type_children = 1;
-    lut_pb_type->modes[1].mode_power = (t_mode_power*)vtr::calloc(1,
-                                                                  sizeof(t_mode_power));
+    lut_pb_type->modes[1].mode_power = new t_mode_power();
     lut_pb_type->modes[1].pb_type_children = new t_pb_type[1];
     alloc_and_load_default_child_for_pb_type(lut_pb_type, default_name,
                                              lut_pb_type->modes[1].pb_type_children);
     /* moved annotations to child so delete old annotations */
     for (i = 0; i < lut_pb_type->num_annotations; i++) {
         for (j = 0; j < lut_pb_type->annotations[i].num_value_prop_pairs; j++) {
-            free(lut_pb_type->annotations[i].value[j]);
+            vtr::free(lut_pb_type->annotations[i].value[j]);
         }
-        free(lut_pb_type->annotations[i].value);
-        free(lut_pb_type->annotations[i].prop);
+        delete[] lut_pb_type->annotations[i].value;
+        delete[] lut_pb_type->annotations[i].prop;
         if (lut_pb_type->annotations[i].input_pins) {
-            free(lut_pb_type->annotations[i].input_pins);
+            vtr::free(lut_pb_type->annotations[i].input_pins);
```

Comment:
Possibly we could convert these C-strings to std::string so we don't need free calls for them, but it depends on how many places they are used (how much code you have to go change to use C++ strings)

Reply:
Same for .output_pins and .clock